cli/provision.go

```
@@ -240,3 +178,97 @@ func NewProvisionCmd() *cobra.Command {
 
 	return &cmd
 }
+
+func thingsFromFile(path string) ([]mfxsdk.Thing, error) {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return []mfxsdk.Thing{}, err
+	}
+
+	file, err := os.OpenFile(path, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		return []mfxsdk.Thing{}, err
+	}
+	defer file.Close()
+
+	things := []mfxsdk.Thing{}
+	switch filepath.Ext(path) {
+	case ".csv":
+		reader := csv.NewReader(file)
+
+		for {
+			l, err := reader.Read()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return []mfxsdk.Thing{}, err
+			}
+
+			if len(l) < 1 {
+				return []mfxsdk.Thing{}, err
```

You should return a custom error since `err` is probably nil here.